test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -220,19 +220,33 @@ session_simulator::commit_transaction(const std::string &config)
         }
     }
 
+    /* We need to rollback a transaction if it failed earlier. */
+    if (_txn_error) {
```

Comment:
We should move this check right after the txn_running check.

Reply:
Maybe even before? If there is an error while creating the transaction, `_txn_running` might not be set to `true`?

Reply:
No, txn_error is not set in begin_transaction, it is set in the calls after begin_transaction, for instance, timestamp_transaction or prepared_transaction. 
I suggest we put this check [here](https://github.com/wiredtiger/wiredtiger/pull/8415/files#diff-5b7ad4d9e453bceb87d0253af93e4429595c436470fda695b483d3b9b1a7b96cR163).

Reply:
Yeah that makes sense, I've moved it.